packages/shared_preferences/shared_preferences_platform_interface/lib/types.dart

```diff
@@ -2,9 +2,55 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-/// Filter options used to get and clear preferences.
+// shared_preferences_async types.
+
+import 'package:flutter/foundation.dart';
+
+/// Basic options for creating SharedPreferencesAsync classes.
+///
+/// This class exists to provide extension to platform specific options as
+/// there are currently no general options that are not platform specific.
+@immutable
+class SharedPreferencesOptions {
+  /// Constructor for SharedPreferencesOptions.
+  const SharedPreferencesOptions();
+}
+
+/// Filter options used to get and clear preferences on shared_preferences_async.
+class PreferencesFilters {
+  /// Creates a new instance with the given options.
+  const PreferencesFilters({
+    this.allowList,
+  });
+
+  /// A list of preference keys that will limit getting and clearing to only
+  /// items included in this list.
```

Comment:
Let's comment about the difference between null and empty.

Reply:
an empty list would break functionality, I think we should have them act the same.

Reply:
We generally don't want two ways of expressing the same thing, collection-wise, so either we should only allow one (usually we do that by making the collection non-nullable, but here I think throwing an argument error on empty might be clearer), or we should make them do different things and just allow the somewhat odd case of empty. It *might* be useful in some circumstances where the set being passed is dynamically generated?

Reply:
That's what I was thinking, if you want to call "nothing" you do want to get everything.